astropy/io/ascii/tests/test_fixedwidth.py

```diff
@@ -239,6 +239,56 @@ def test_read_detect_col_starts_or_ends():
 dat = ascii.read(table, Reader=ascii.FixedWidth)
 
 
+def test_read_header_col_onesided():
+    """Table with only start or end positions of columns demarked by header"""
+    table = "\n".join([' Col1   Col2         Col3        Col4      ',
+                       '  1.2    "hello"      1           a         ',
```

Comment:
If you use float values that are exactly represented, for instance 1.0 or 1.5 then the equality testing can be simplified to use `np.array_equal` always.